test/safe_pg_migrations_test.rb

```diff
@@ -315,6 +316,166 @@ def change
+        'NOT VALID',
+      "SET statement_timeout TO '70s'",
+    ], calls
+  end
+
+  def test_add_foreign_key
+    @connection.create_table(:users) { |t| t.string :email }
+    @connection.create_table(:messages) do |t|
+      t.string :message
+      t.bigint :user_id
+    end
+
+    @migration =
+      Class.new(ActiveRecord::Migration::Current) do
+        def change
+          add_foreign_key :messages, :users
+        end
+      end.new
+
+    calls = record_calls(@connection, :execute) { run_migration }
+    assert_calls [
+      "SET statement_timeout TO '5s'",
+      'ALTER TABLE "messages" ADD CONSTRAINT "fk_rails_273a25a7a6" FOREIGN KEY ("user_id") REFERENCES "users" ("id") ' \
+        'NOT VALID',
+      "SET statement_timeout TO '70s'",
+      'SET statement_timeout TO 0',
+      'ALTER TABLE "messages" VALIDATE CONSTRAINT "fk_rails_273a25a7a6"',
+      "SET statement_timeout TO '70s'",
+    ], calls
+  end
+
+  def test_add_foreign_key_idem_potent
+    @connection.create_table(:users) { |t| t.string :email }
+    @connection.create_table(:messages) do |t|
+      t.string :message
+      t.bigint :user_id
+    end
+
+    @migration =
+      Class.new(ActiveRecord::Migration::Current) do
+        def change
+          2.times { add_foreign_key :messages, :users }
+        end
+      end.new
+
+    execute_calls = nil
+    write_calls =
+        record_calls(@migration, :write) do
+          execute_calls = record_calls(@connection, :execute) { run_migration }
+        end
+    assert_calls [
+      "SET statement_timeout TO '5s'",
+      'ALTER TABLE "messages" ADD CONSTRAINT "fk_rails_273a25a7a6" FOREIGN KEY ("user_id") REFERENCES "users" ("id") ' \
+        'NOT VALID',
+      "SET statement_timeout TO '70s'",
+      'SET statement_timeout TO 0',
+      'ALTER TABLE "messages" VALIDATE CONSTRAINT "fk_rails_273a25a7a6"',
+      "SET statement_timeout TO '70s'",
+      "SET statement_timeout TO '5s'",
+      "SET statement_timeout TO '70s'",
+      "SET statement_timeout TO 0",
+      'ALTER TABLE "messages" VALIDATE CONSTRAINT "fk_rails_273a25a7a6"',
+      "SET statement_timeout TO '70s'",
+    ], execute_calls
+
+    assert_equal [
+      "== 8128 : migrating ===========================================================",
+      "-- add_foreign_key(:messages, :users)",
+      ], write_calls.map(&:first)[0...2]
+    assert_equal [
+      "-- add_foreign_key(:messages, :users)",
+      "   -> /!\\ Foreign key 'messages' -> 'users' already exists. Skipping statement."
+      ], write_calls.map(&:first)[3...5]
+  end
+
+  def test_add_foreign_key_idem_potent_with_column_option
+    @connection.create_table(:users) { |t| t.string :email }
+    @connection.create_table(:messages) do |t|
+      t.string :message
+      t.bigint :user_id
+    end
+
+    @migration =
+      Class.new(ActiveRecord::Migration::Current) do
+        def change
+          2.times { add_foreign_key :messages, :users, column: :user_id }
```

Comment:
you should user another naming for the column, `user_id` is the default one, go for `author_id`